storage/innobase/os/os0proc.cc

```diff
@@ -155,19 +188,39 @@ os_mem_alloc_large(
 	ut_ad(ut_is_2pow(size));
 	size = *n = ut_2pow_round(*n + (size - 1), size);
 	ptr = mmap(NULL, size, PROT_READ | PROT_WRITE,
-		   MAP_PRIVATE | OS_MAP_ANON, -1, 0);
+		   MAP_PRIVATE | OS_MAP_ANON |
+		   (populate ? OS_MAP_POPULATE : 0), -1, 0);
 	if (UNIV_UNLIKELY(ptr == (void*) -1)) {
```

Comment:
Although this is not your changes, the return value of the `mmap` should be compared with `MAP_FAILED` rather than `-1`